networkx/algorithms/centrality/group.py

```diff
@@ -84,49 +84,196 @@ def group_betweenness_centrality(G, C, normalized=True, weight=None):
        The Centrality of Groups and Classes.
        Journal of Mathematical Sociology. 23(3): 181-201. 1999.
        http://www.analytictech.com/borgatti/group_centrality.htm
-    .. [2] Ulrik Brandes:
-       On Variants of Shortest-Path Betweenness
-       Centrality and their Generic Computation.
-       Social Networks 30(2):136-145, 2008.
-       http://citeseerx.ist.psu.edu/viewdoc/download?doi=10.1.1.72.9610&rep=rep1&type=pdf
-    .. [3] Sourav Medya et. al.:
-       Group Centrality Maximization via Network Design.
-       SIAM International Conference on Data Mining, SDM 2018, 126–134.
-       https://sites.cs.ucsb.edu/~arlei/pubs/sdm18.pdf
+    .. [2] Rami Puzis, Yuval Elovici, and Shlomi Dolev.
+       "Fast algorithm for successive computation of group betweenness centrality."
+        https://journals.aps.org/pre/pdf/10.1103/PhysRevE.76.056709?casa_token=YvH6xTIc6AUAAAAA%3A90BkxMRiQyOg9y7PEU_lrcf1FXPF_sqTCI_Y_2lD0QodSeyXAQtz-s0_e1Z0CtkZ1bWdVCkK9S5fXw
+
     """
-    betweenness = 0  # initialize betweenness to 0
+    GBC = []  # initialize betweenness
     V = set(G)  # set of nodes in G
-    C = set(C)  # set of nodes in C (group)
-    if len(C - V) != 0:  # element(s) of C not in V
-        raise nx.NodeNotFound(
-            "The node(s) " + str(list(C - V)) + " are not " "in the graph."
-        )
-    V_C = V - C  # set of nodes in V but not in C
-    # accumulation
-    for pair in combinations(V_C, 2):  # (s, t) pairs of V_C
-        try:
-            paths = 0
-            paths_through_C = 0
-            for path in nx.all_shortest_paths(
-                G, source=pair[0], target=pair[1], weight=weight
-            ):
-                if set(path) & C:
-                    paths_through_C += 1
-                paths += 1
-            betweenness += paths_through_C / paths
-        except nx.exception.NetworkXNoPath:
-            betweenness += 0
-    # rescaling
-    v, c = len(G), len(C)
-    if normalized:
-        scale = 1 / ((v - c) * (v - c - 1))
-        if not G.is_directed():
-            scale *= 2
-    else:
-        scale = None
-    if scale is not None:
-        betweenness *= scale
-    return betweenness
+
+    #  check weather C contains one or many groups
+    if not any(isinstance(el, list) for el in C):
```

Comment:
this if statment doesn't work if C is a list of lists. I use this instead. Do you another idea?
```suggestion
    if not any(isinstance(el, list) for el in C) and not any(isinstance(el, set) for el in C):
```

Reply:
I thought this one does work for a list of lists... But maybe I am misunderstanding. 

If C is a list of lists or a set of sets (or a list of sets or a set of lists)  -> container of containers... Then C is good to go and we don't do anything.  If C is a collection of nodes, then we put that inside a list to make it a list of a single collection.  Do I have this part right?

So, what we want to check for is whether we have a collection of nodes. 
`if all(el in V for el in C):` does that. But again, maybe I am not understanding... What doesn't work here?

Reply:
If C is a list of lists  (or a set of lists) then we get `TypeError: unhashable type: 'list'`.
On second though I change el to be always a set: `if all(set(el) in V for el in C):`

Reply:
Hi,
two problems with this approach:
1) C may have nodes not in the graph (like the test test_group_betweenness_node_not_in_graph).
2) C may be list of lists or set of lists and then we receive  `TypeError: unhashable type: 'list'`

Reply:
Use `if all(el in G for el in C):` to avoid that `TypeError:` in case 2). Using `in G` instead of `in V` doesn't raise the TypeError. There isn't really any reason to have `V` defined with the new code. Let's just not define V. We can use `in G` to check for a node, and set operations can use `G.nodes`.

What do you think of checking whether the first element of C is in the Graph. If it is then we treat C as a group.  If the first element isn't in the Graph, it is presumably a container of nodes and we can assume the rest of the elements of C are containers of nodes as well. There is probably some way that this messes up too.  But I can't see it.

```python
thing = nx.utils.arbitrary_element(C)
if thing in g:
    C = [C]
nodes = {node for node in group for group in C}
if nodes - G.nodes:  # elements(s) of C are not in G
    raise nx.NodeNotFound(f"The node(s) {nodes - G.nodes} are in C but not in G")
```

Reply:
I suggest something like this: `if any(el in G for el in C):`. If C has one element which is a node in the graph then we treat C as a group. The one way to mess it up would be to have every element in C not in G. In that case we would get a `TypeError: 'int' object is not iterable`. @dschult what do you think about that?